namer/consul/src/main/scala/io/buoyant/namer/consul/LookupCache.scala

```diff
@@ -11,24 +12,31 @@ import io.buoyant.consul.v1
 private[consul] class LookupCache(
   consulApi: v1.ConsulApi,
   agentApi: v1.AgentApi,
-  setHost: Boolean = false
+  setHost: Boolean = false,
+  stats: StatsReceiver = NullStatsReceiver
 ) {
 
+  private[this] val negs = stats.counter("negs")
+  private[this] val leafs = stats.counter("leafs")
+
   def apply(
     dc: String,
     key: SvcKey,
     id: Path,
     residual: Path
   ): Activity[NameTree[Name]] = {
     log.debug("consul lookup: %s %s", id.show)
+
     Dc.watch(dc).map { services =>
       services.get(key) match {
         case None =>
           log.debug("consul dc %s service %s missing", dc, key)
+          negs.incr()
```

Comment:
These stats are potentially confusing.  Since they're in the map block of an activity, they will get incremented for each listener registered to this activity.  Therefore the counter represents something like "number of negs returned" rather than "total number of negs that exist" or something that one might expect.